broadcast/internal/arc/arc_submit_tx.go

```diff
@@ -139,21 +143,39 @@ func submitBatchTransactions(ctx context.Context, arc *ArcClient, txs []*broadca
 	return model, nil
 }
 
-func createSubmitTxBody(tx *broadcast.Transaction) ([]byte, error) {
-	body := &SubmitTxRequest{tx.RawTx}
-	data, err := json.Marshal(body)
+func createSubmitTxBody(arc *ArcClient, tx *broadcast.Transaction, txFormat broadcast.TransactionFormat) ([]byte, error) {
+	body := &SubmitTxRequest{tx.Hex}
+
+	if txFormat == broadcast.RawTxFormat {
+		if err := convertToEfTransaction(arc, body); err != nil {
+			return nil, fmt.Errorf("Conversion to EF format failed: %s", err.Error())
+		}
+	} else if txFormat == broadcast.BeefFormat {
+		// To be implemented
+		return nil, fmt.Errorf("Submitting transactions in BEEF format is unimplemented yet...")
+	}
 
+	data, err := json.Marshal(body)
 	if err != nil {
 		return nil, ErrSubmitTxMarshal
 	}
 
 	return data, nil
 }
 
-func createSubmitBatchTxsBody(txs []*broadcast.Transaction) ([]byte, error) {
+func createSubmitBatchTxsBody(arc *ArcClient, txs []*broadcast.Transaction, txFormat broadcast.TransactionFormat) ([]byte, error) {
 	rawTxs := make([]*SubmitTxRequest, 0, len(txs))
 	for _, tx := range txs {
-		rawTxs = append(rawTxs, &SubmitTxRequest{RawTx: tx.RawTx})
+		rawTxs = append(rawTxs, &SubmitTxRequest{RawTx: tx.Hex})
```

Comment:
Maybe the approach switch could be extracted further 
and then instead of having 2 loops, you could just do one loop like:
```go
	for _, tx := range txs {
		rawTxs = append(rawTxs, createSubmitTxRequest(tx))
	}
```